Model/User.hs

```diff
@@ -45,11 +53,32 @@ upsertUser user = entityKey <$> upsert user
     , UserEmail =. userEmail user
     ]
 
-credsToUser :: Maybe PlanId -> Creds m -> Maybe User
+credsToUser :: Maybe PlanId -> Creds m -> Either String User
 credsToUser mplanId Creds{..} = User
-    <$> lookup "name" credsExtra
-    <*> lookup "email" credsExtra
+    <$> (profileName <$> eprofile)
+    <*> (profileEmail <$> eprofile)
     <*> pure credsPlugin
     <*> pure credsIdent
-    <*> mplanId
+    <*> eplanId
     <*> pure Nothing
+
+  where
+    eplanId = maybe (Left "Free plan not found") Right mplanId
+    eprofile = extraToProfile credsPlugin credsExtra
+
+extraToProfile :: Text -> [(Text, Text)] -> Either String Profile
+extraToProfile "dummy" _ = Right $ dummyProfile
+extraToProfile "github" extra = githubProfile extra
+extraToProfile plugin _ = Left $ "Invalid plugin: " ++ unpack plugin
+
+dummyProfile :: Profile
+dummyProfile = Profile "Dummy Login" "dummy@example.com"
+
+githubProfile :: [(Text, Text)] -> Either String Profile
+githubProfile extra = Profile
```

Comment:
Nope, this is an Applicative expression. `Profile` is combined with the two lookups via `(<$>)` and `(<*>)`. Because of the semantics of those operators for `Either e`, if both are found you get a `Right profile` otherwise a `Left error`.